src/Components/Components/src/RenderTree/Renderer.cs

```diff
@@ -956,10 +955,31 @@ private void HandleExceptionViaErrorBoundary(Exception error, ComponentState? er
         /// Releases all resources currently used by this <see cref="Renderer"/> instance.
         /// </summary>
         /// <param name="disposing"><see langword="true"/> if this method is being invoked by <see cref="IDisposable.Dispose"/>, otherwise <see langword="false"/>.</param>
-        protected virtual void Dispose(bool disposing)
+        protected virtual void Dispose(bool disposing) => DisposeAsync().AsTask().GetAwaiter().GetResult();
```

Comment:
@javiercn you have the most context here - do you remember why `Dispose` doesn't implement by calling `DisposeAsync` and blocking?

Reply:
Likely because that was virtual and would break consumers

Reply:
Blocking is impossible and throws on WebAssembly. I know you don't normally dispose a renderer on WebAssembly but nothing stops you.